google-cloud-spanner/src/main/java/com/google/cloud/spanner/Database.java

```diff
@@ -194,17 +194,59 @@ private String database() {
   static Database fromProto(
       com.google.spanner.admin.database.v1.Database proto, DatabaseAdminClient client) {
     checkArgument(!proto.getName().isEmpty(), "Missing expected 'name' field");
-    return new Database.Builder(client, DatabaseId.of(proto.getName()))
-        .setState(fromProtoState(proto.getState()))
-        .setCreateTime(Timestamp.fromProto(proto.getCreateTime()))
-        .setRestoreInfo(RestoreInfo.fromProtoOrNullIfDefaultInstance(proto.getRestoreInfo()))
-        .setVersionRetentionPeriod(proto.getVersionRetentionPeriod())
-        .setEarliestVersionTime(Timestamp.fromProto(proto.getEarliestVersionTime()))
-        .setEncryptionConfig(CustomerManagedEncryption.fromProtoOrNull(proto.getEncryptionConfig()))
-        .setDefaultLeader(proto.getDefaultLeader())
-        .setDialect(Dialect.fromProto(proto.getDatabaseDialect()))
-        .setProto(proto)
-        .build();
+    DatabaseInfo.Builder builder =
+        new Builder(client, DatabaseId.of(proto.getName()))
+            .setState(fromProtoState(proto.getState()))
+            .setCreateTime(Timestamp.fromProto(proto.getCreateTime()))
+            .setRestoreInfo(RestoreInfo.fromProtoOrNullIfDefaultInstance(proto.getRestoreInfo()))
+            .setVersionRetentionPeriod(proto.getVersionRetentionPeriod())
+            .setEarliestVersionTime(Timestamp.fromProto(proto.getEarliestVersionTime()))
+            .setEncryptionConfig(
+                CustomerManagedEncryption.fromProtoOrNull(proto.getEncryptionConfig()))
+            .setDefaultLeader(proto.getDefaultLeader())
+            .setDialect(Dialect.fromProto(proto.getDatabaseDialect()))
+            .setReconciling(proto.getReconciling())
+            .setProto(proto);
+    if (proto.getEnableDropProtection()) {
+      builder.enableDropProtection();
+    } else {
+      builder.disableDropProtection();
+    }
+    return builder.build();
+  }
+
+  public com.google.spanner.admin.database.v1.Database toProto() {
+    if (getProto() != null) {
+      return getProto();
+    }
```

Comment:
I don't think we should have this here. 
What if someone gets a com.google.cloud.spanner.Database object from the backend (after fromProto conversion), and then user updates specific fields in the same object, and tries to make RPC calls with the corresponding proto object? toProto() will still return the old proto object.